--- slock.c	2008-07-29 13:22:46.000000000 -0500
+++ ../slock.c	2010-10-15 07:13:34.893155713 -0500
@@ -125,10 +125,10 @@
 
 	/* main event loop */
 	while(running && !XNextEvent(dpy, &ev)) {
-		if(len == 0 && DPMSCapable(dpy)) {
-			DPMSEnable(dpy);
-			DPMSForceLevel(dpy, DPMSModeOff);
-		}
+		/*if(len == 0 && DPMSCapable(dpy)) {*/
+			/*DPMSEnable(dpy);*/
+			/*DPMSForceLevel(dpy, DPMSModeOff);*/
+		/*}*/
 		if(ev.type == KeyPress) {
 			buf[0] = 0;
 			num = XLookupString(&ev.xkey, buf, sizeof buf, &ksym, 0);
